recirq/quantum_chess/initial_mapping_utils_test.py

```diff
@@ -0,0 +1,386 @@
+            grid_qubits['0_2'],
+            grid_qubits['1_3'],
+            grid_qubits['0_4'],
+        ],
+        grid_qubits['0_4']: [
+            grid_qubits['0_3'],
+            grid_qubits['1_4'],
+            grid_qubits['0_5'],
+        ],
+        grid_qubits['0_5']: [
+            grid_qubits['0_4'],
+            grid_qubits['1_5'],
+            grid_qubits['0_6'],
+        ],
+        grid_qubits['0_6']: [
+            grid_qubits['0_5'],
+            grid_qubits['1_6'],
+            grid_qubits['0_7'],
+        ],
+        grid_qubits['0_7']: [
+            grid_qubits['0_6'],
+            grid_qubits['1_7'],
+            grid_qubits['0_8'],
+        ],
+        grid_qubits['0_8']: [
+            grid_qubits['0_7'],
+            grid_qubits['1_8'],
+            grid_qubits['0_9'],
+        ],
+        grid_qubits['0_9']: [
+            grid_qubits['0_8'],
+            grid_qubits['1_9'],
+            grid_qubits['0_10'],
+        ],
+        grid_qubits['0_10']: [
+            grid_qubits['0_9'],
+            grid_qubits['1_10'],
+        ],
+        grid_qubits['1_0']: [
+            grid_qubits['1_1'],
+            grid_qubits['0_0'],
+        ],
+        grid_qubits['1_1']: [
+            grid_qubits['1_0'],
+            grid_qubits['0_1'],
+            grid_qubits['1_2'],
+        ],
+        grid_qubits['1_2']: [
+            grid_qubits['1_1'],
+            grid_qubits['0_2'],
+            grid_qubits['1_3'],
+        ],
+        grid_qubits['1_3']: [
+            grid_qubits['1_2'],
+            grid_qubits['0_3'],
+            grid_qubits['1_4'],
+        ],
+        grid_qubits['1_4']: [
+            grid_qubits['1_3'],
+            grid_qubits['0_4'],
+            grid_qubits['1_5'],
+        ],
+        grid_qubits['1_5']: [
+            grid_qubits['1_4'],
+            grid_qubits['0_5'],
+            grid_qubits['1_6'],
+        ],
+        grid_qubits['1_6']: [
+            grid_qubits['1_5'],
+            grid_qubits['0_6'],
+            grid_qubits['1_7'],
+        ],
+        grid_qubits['1_7']: [
+            grid_qubits['1_6'],
+            grid_qubits['0_7'],
+            grid_qubits['1_8'],
+        ],
+        grid_qubits['1_8']: [
+            grid_qubits['1_7'],
+            grid_qubits['0_8'],
+            grid_qubits['1_9'],
+        ],
+        grid_qubits['1_9']: [
+            grid_qubits['1_8'],
+            grid_qubits['0_9'],
+            grid_qubits['1_10'],
+        ],
+        grid_qubits['1_10']: [
+            grid_qubits['1_9'],
+            grid_qubits['0_10'],
+        ],
+    }
+    assert len(g) == len(expected)
+    for q in expected:
+        assert set(g[q]) == set(expected[q])
+
+
+def test_get_least_connected_qubit():
+    g = {
+        0: [1],
+        1: [0, 2],
+        2: [1],
+        3: [4],
+        4: [3],
+    }
+    assert imu.get_least_connected_qubit(g, deque([0, 1, 2])) in {0, 2}
+    assert imu.get_least_connected_qubit(g, deque([3, 4])) in {3, 4}
```

Comment:
Do we want to have a test case that is less ambiguous?

Reply:
added!